frontend/src/metabase/visualizations/lib/settings/graph.js

```diff
@@ -198,19 +202,22 @@ export const GRAPH_DATA_SETTINGS = {
         .filter(vizSettings["graph._metric_filter"])
         .map(getOptionFromColumn);
 
+      const addedMetrics = vizSettings["graph.metrics"];
       const hasBreakout = vizSettings["graph.dimensions"].length > 1;
-      const addedMetricsCount = vizSettings["graph.metrics"].length;
+      const addedMetricsCount = addedMetrics.length;
       const maxMetricsSupportedCount = getMaxMetricsSupported(card.display);
 
       const hasMetricsToAdd = options.length > addedMetricsCount;
       const canAddAnother =
         addedMetricsCount < maxMetricsSupportedCount &&
         hasMetricsToAdd &&
-        !hasBreakout;
+        !hasBreakout &&
+        addedMetrics.filter(metric => metric === null || metric === undefined)
+          .length === 0;
```

Comment:
nit: I think we should use array APIs to explicitly communicate this instead of checking the length which requires readers to have to think twice (filter + length). Also, we could shorten the condition using double equal (or not equal in this case) instead

```suggestion
        addedMetrics.every(metric => metric != null)
```

What do you think?

Reply:
That does look cleaner. I'll change it